src/simple_commander/controllers/main.py

```diff
@@ -22,169 +26,189 @@
-            self.fire(unit)
-            return True
-        elif self.bullet.y < 0 or self.bullet.y > game_field_height:
-            return True
-        else:
-            return False
-
-    def fire(self, other_unit):
+    def rotate(self, angle):
+        self.angle += angle
+
+    def change_speed(self, speed):
+        new_speed = self.speed + speed
+        self.speed = new_speed > 0 and new_speed or 0
+
+    def check_collision(self, other_unit, all_units):
+        # check if coordinate for two units is the same
+        # for this check we also include width and height of unit's image
+        # (other_unit.x - other_unit.width / 2 < self.x < other_unit.x + other_unit.width / 2)
+        # (other_unit.y - other_unit.height / 2 < self.y < other_unit.y + other_unit.height / 2)
+        if id(self) != id(other_unit) and getattr(self, 'unit_id', '') != id(other_unit) and \
+                getattr(other_unit, 'unit_id', '') != id(self):
+            if (self.x > other_unit.x - other_unit.width / 2) and (self.x < other_unit.x + other_unit.width / 2) and \
+                    (self.y > other_unit.y - other_unit.height / 2) and (self.y < other_unit.y + other_unit.height / 2):
+                self.kill(other_unit, all_units)
+
+    def reset(self):
         raise NotImplementedError
 
-    def move_bullet(self):
+    def kill(self, other_unit, units):
         raise NotImplementedError
 
 
 class Invader(Unit):
-    moving_speed = 0
 
-    def __init__(self, x, y, bonus=0, speed=5, unit_filename='',
+    def __init__(self, x, y, angle, bonus=10, speed=DEFAULT_SPEED, unit_filename='',
                  bullet_filename=IMAGE_FILENAME.get('bullet_invader', '')):
         if not unit_filename and len(IMAGE_FILENAME.get('invader', [])):
             random_number = randint(0, len(IMAGE_FILENAME.get('invader', [])) - 1)
             unit_filename = IMAGE_FILENAME.get('invader', [])[random_number]
-        super(Invader, self).__init__(x, y, bonus, speed, unit_filename, bullet_filename)
-        self.moving_speed = speed
-
-    def check_if_move_y(self):
-        return self.x <= self.shift
-
-    def set_speed(self, game_field_width):
-        # this check for first Invader
-        if self.x <= self.shift:
-            self.moving_speed = self.speed or -self.speed  # positive value
-        # this check for last Invader
-        if self.x >= game_field_width:
-            self.moving_speed = self.speed < 0 and self.speed or -self.speed  # negative value
-
-    def fire(self, hero):
-        if hero.life_count > 1:
-            hero.life_count -= 1
+        super(Invader, self).__init__(x, y, angle, bonus, speed, unit_filename, bullet_filename)
+
+    def reset(self):
+        self.angle = randint(0, 360)
+        self.step = 0
+        self.x0 = self.x
+        self.y0 = self.y
+
+    def kill(self, other_unit, units):
+        unit_class_name = other_unit. __class__.__name__
```

Comment:
Can you add logging on each event. It must be very helpful for debugging and game units tracking